agent/src/beerocks/slave/son_slave_thread.cpp

```diff
@@ -3568,10 +3568,12 @@ bool slave_thread::slave_fsm(bool &call_slave_select)
     case STATE_BACKHAUL_MANAGER_CONNECTED: {
         LOG(TRACE) << "MASTER_CONNECTED";
 
+        master_socket = backhaul_manager_socket;
+        master_socket->setPeerMac(backhaul_params.controller_bridge_mac);
```

Comment:
Are you sure this has no effect? When `wlan_settings.band_enabled` is 0, `master_socket` was never set, and now you always set it.

Reply:
The opposite, if the band enable is 0 the master socket was set immediately.
Basically, the cases in which the socket is not set are on all the cases we do slave_reset(), which we unset it there anyway. So if we didn't go to slave reset, we need to set it.